scipy/stats/arb_dist.py

```diff
@@ -0,0 +1,269 @@
+#
+# Author:  Andrew Nelson 2016
+# arbitrary distribution
+#
+from functools import partial
+import numpy as np
+from scipy.stats import rv_continuous
+from scipy.integrate import quad
+from scipy.interpolate import InterpolatedUnivariateSpline, interp1d
+
+from numpy.testing import (assert_equal, TestCase, assert_allclose,
+                           run_module_suite, assert_almost_equal,
+                           assert_string_equal, assert_raises, assert_)
+
+class rv_arbitrary(rv_continuous):
+    r"""
+    An arbitrary continuous random variable.
+
+    Parameters
+    ----------
+    y : np.array or callable
+        Specifies the probability distribution function. Can be specified as an
+        array or as a function returning a single value. If `y` is an array
+        then the distribution is a sampled PDF and should have a corresponding
+        `x` value. Note that the area under the PDF is normalised to 1, i.e.
+        ``rv_arbitrary.cdf(np.max(x)) == 1``. All values in `y` should be
+        ``y >= 0``.
+    x : array_like
+        Range of support for the distribution. If `y` is a function then the
+        distribution is defined over ``[np.min(x), np.max(x)]``. If `y` is an
```

Comment:
``x`` should already be a sequence of length 2 in this case, I wouldn't allow longer input for clarity.